var/spack/repos/builtin/packages/zoltan/package.py

```diff
@@ -45,13 +43,36 @@ class Zoltan(Package):
 
     conflicts('+parmetis', when='~mpi')
 
-    def install(self, spec, prefix):
+    build_directory = 'spack-build'
+
+    @property
+    def configure_directory(self):
+        spec = self.spec
+
         # FIXME: The older Zoltan versions fail to compile the F90 MPI wrappers
         # because of some complicated generic type problem.
         if spec.satisfies('@:3.6+fortran+mpi'):
             raise RuntimeError(('Cannot build Zoltan v{0} with +fortran and '
                                 '+mpi; please disable one of these features '
                                 'or upgrade versions.').format(self.version))
+        if spec.satisfies('@:3.6'):
+            zoltan_path = 'Zoltan_v{0}'.format(self.version)
+            return zoltan_path
+        return '.'
+
+    @property
+    def parallel(self):
+        # NOTE: Earlier versions of Zoltan cannot be built in parallel
+        # because they contain nested Makefile dependency bugs.
+        return not self.spec.satisfies('@:3.6+fortran')
+
+    def autoreconf(self, spec, prefix):
+        autoreconf = which('autoreconf')
```

Comment:
This line requires the addition of appropriate `depends_on()` directives so a Spack-built `autoreconf` is available.
```
depends_on('autoconf', type='build', when='@develop')
depends_on('automake', type='build', when='@develop')
depends_on('libtool',  type='build', when='@develop')
depends_on('m4',       type='build', when='@develop')
```

Reply:
I add depends_on().
I also add to use old version perl for old version, because the perl script in the old source use defined(@ARGV).